src/app/app.rs

```diff
@@ -107,6 +108,58 @@ impl App {
             watcher.watch(Path::new(path), RecursiveMode::Recursive).unwrap();
         };
 
+        // audio
+
+        fn silence(_:f32) -> (f32, f32) {(0.0, 0.0)}
+        
+        let sf = StreamFactory::default_factory().unwrap();
+
+        let sample_rate = sf.config().sample_rate.0;
+        let mut sample_clock = 0;
+        let routin = move |len: usize| -> Vec<f32> {
+            (0..len / 2) // len is apparently left *and* right
+                .flat_map(|_| {
+                    sample_clock = (sample_clock + 1) % sample_rate;
+
+                    let (l, r) = audio_shader.unwrap_or(silence)(sample_clock as f32 / sample_rate as f32);
+                    vec![l, r]
+                })
+                .collect()
+        };
+        
+        let stream = sf.create_stream(routin).unwrap(); // creates stream from function "routin"
+        StreamTrait::play(&stream).unwrap();
+
+        /* For some reason if it's possible to branch, it doesn't play any sound
+
+        let mut play_audio = true;
```

Comment:
Since play_audio is initialized on true, and never set to false it would indeed always play audio. To only execute code when an Option contains data we can use `if let Some(x)` to do that :) https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html